src/main/scala/beam/agentsim/infrastructure/ChargingNetwork.scala

```diff
@@ -147,7 +151,7 @@ object ChargingNetwork {
       vehicle: BeamVehicle,
       stall: ParkingStall,
       theSender: ActorRef
-    ): ChargingVehicle = this.synchronized {
+    ): ChargingVehicle = {
```

Comment:
These synchronizations were likely added to make sure the data did not get corrupted...we should leave these. Unless you are going to convert this into an actor or something else which can handle parallel reads/writes.

Reply:
You are right, this makes this class not thread-safe, but this is the price for a better performance. Though I think that there's no big difference in performance. So I just restored these synchronizations.